registry/storage/paths.go

```diff
@@ -491,3 +530,33 @@ func digestPathComponents(dgst digest.Digest, multilevel bool) ([]string, error)
 
 	return append(prefix, suffix...), nil
 }
+
+// Reconstructs a digest from a path which must end in "data"
+func digestFromPath(digestPath string, multilevel bool) (digest.Digest, error) {
+	pathComponents := strings.Split(digestPath, "/")
+	var minLength int
+	if multilevel {
```

Comment:
This is bazaar. Can you find a better way to do this lookup? We already have pathSpecs and other methods that do this. Please use those.